modules/layers/src/arc-layer/arc-layer-uniforms.ts

```diff
@@ -0,0 +1,39 @@
+import {ShaderModule} from '@luma.gl/shadertools';
+import {UniformTypes} from '@deck.gl/core';
+
+const uniformBlock = `\
+uniform arcUniforms {
+  bool greatCircle;
+  bool useShortestPath;
+  float numSegments;
+  float widthScale;
+  float widthMinPixels;
+  float widthMaxPixels;
+  highp int widthUnits;
+} arc;
+`;
+
+export type ArcProps = {
+  greatCircle: boolean;
+  useShortestPath: boolean;
+  numSegments: number;
+  widthScale: number;
+  widthMinPixels: number;
+  widthMaxPixels: number;
+  widthUnits: number;
+};
+
+export const arcUniforms = {
```

Comment:
I find this separation into a module for just the uniforms a bit artificial.  

Why not go all the way and just package the shaders (or at least the bulk of them) into this module as well. 

I think that could be an opportunity for a second pass.

Reply:
I actually quite like the fact that this quite verbose and unsurprising mapping code is hidden in a separate file. Agree we could consider moving the shaders for a second pass, but I worry that it could make it more complex for custom layers to inject/modify code